action_server/src/sema4ai/action_server/_storage.py

```diff
@@ -0,0 +1,159 @@
+import base64
+import os
+import typing
+from functools import lru_cache
+from logging import getLogger
+from pathlib import Path
+
+from pydantic.main import BaseModel
+
+from sema4ai.action_server._settings import get_default_settings_dir
+
+log = getLogger(__name__)
+
+STORAGE_FILE_NAME = "action_server_storage.json"
+STORAGE_PATH = Path(os.path.join(get_default_settings_dir(), STORAGE_FILE_NAME))
```

Comment:
`get_default_settings_dir` should not be called at user level (it has a side-effect of creating a directory). This means that instead of `STORAGE_PATH` being a constant it should be a function such as `get_storage_path`.